tests/v3/test_group.py

```diff
@@ -18,6 +19,7 @@
 from zarr.store import StorePath
 
 
+@pytest.mark.skipif(IS_WASM, reason="Can't test async code in WASM")
```

Comment:
This is going to be a problem for moving this PR forward. The entire design of V3 depends on being able to execute async. If threading and async are a show stopper, I think we should evaluate whether this can work another way.

Reply:
From the latest run on my fork:

```pwsh
14 failed, 1606 passed, 1572 skipped, 9 xfailed, 93 warnings in 104.70s (0:01:44)
```

Reply:
I assume 99% of the passing tests are coming from the `v2` test suite. I wouldn't put much weight on that. We need `tests/v3` to be passing for this to work.

Reply:
Yes. If there is a way to run asynchronous tests in a manner that is not asynchronous, then there might be a way forward, but if Zarr's underlying functionality is going to rely on this, I am not sure if there is a solution. The first way would be to get `pytest-asyncio` both packaged and running under Pyodide.

Reply:
It is at this point possible to run async tests in Pyodide using stack switching. Just have to implement it.

Reply:
Thanks for pushing this forward @agriyakhetarpal.